scripts/broad_dsde_workflows/GenotypeGVCFsScatterWf_170204.wdl

```diff
@@ -0,0 +1,173 @@
+## licensing information pertaining to the included programs.
+
+# TASK DEFINITIONS
+
+# Unzip GVCFs because GenotypeGVCFs is picky
+task UnzipGVCF {
+    File gzipped_gvcf
+    String unzipped_basename
+    Int disk_size
+    String mem_size
+
+    # HACK ALERT! Using .gvcf extension here to force IndexFeatureFile to make the right 
+    # kind of index, but afterward we need to change to .g.vcf which is the correct 
+    # for GVCFs.
+    command <<<
+	gunzip -c ${gzipped_gvcf} > ${unzipped_basename}.gvcf
+    java -Xmx2g -jar /usr/gitc/GATK4.jar IndexFeatureFile -F ${unzipped_basename}.gvcf
+    mv ${unzipped_basename}.gvcf ${unzipped_basename}.g.vcf
+    mv ${unzipped_basename}.gvcf.idx ${unzipped_basename}.g.vcf.idx
+    >>>
+
+	runtime {
+	    docker: "broadinstitute/genomes-in-the-cloud:2.2.4-1469632282"
+	    memory: mem_size 
+	    disks: "local-disk " + disk_size + " HDD"
+	}
+
+    output {
+	    File unzipped_gvcf = "${unzipped_basename}.g.vcf"
+	    File gvcf_index = "${unzipped_basename}.g.vcf.idx"
+    }
+}
+
+# Perform joint-genotyping
+task GenotypeGVCFs { 
+	Array[File] gvcfs
+    Array[File] gvcf_indices
+    String vcf_basename
+    File ref_dict
+    File ref_fasta
+    File ref_fasta_index
+  	File interval_list
+    Int disk_size
+    String mem_size
+
+	command {
+		java -XX:GCTimeLimit=50 -XX:GCHeapFreeLimit=10 -Xmx8000m \
+        	-jar /usr/gitc/GATK36.jar \
+        	-T GenotypeGVCFs \
+        	-R ${ref_fasta} \
+        	--variant ${sep=' --variant ' gvcfs} \
+        	-L ${interval_list} \
+        	-o ${vcf_basename}.vcf.gz 
+	}
+
+	output {
+		File genotyped_vcf = "${vcf_basename}.vcf.gz"
+		File genotyped_index = "${vcf_basename}.vcf.gz.tbi"
+	}
+
+	runtime {
+		docker: "broadinstitute/genomes-in-the-cloud:2.2.4-1469632282"
+		memory: mem_size
+    	cpu: "1"
+    	disks: "local-disk " + disk_size + " HDD"
+	}
+}
+
+# Combine multiple VCFs 
+task MergeVCFs {
+    Array [File] input_vcfs
+    Array [File] input_vcfs_indices
+    String vcf_name
+    String vcf_index
+    Int disk_size
+    String mem_size
+
+    command {
+	    java -Xmx2g -jar /usr/gitc/picard.jar \
+	    MergeVcfs \
+	    INPUT=${sep=' INPUT=' input_vcfs} \
+	    OUTPUT=${vcf_name}
+    }
+
+  	runtime {
+	    docker: "broadinstitute/genomes-in-the-cloud:2.2.4-1469632282"
+	    memory: mem_size
+	    disks: "local-disk " + disk_size + " HDD"
+	}
+
+    output {
+    	File output_vcf = "${vcf_name}"
+    	File output_vcf_index = "${vcf_index}"
+    }
+}
+
+workflow GenotypeGVCFsScatterWf {
+  	File ref_fasta
+  	File ref_fasta_index
+  	File ref_dict
+	Array[File] input_gvcfs
+	Array[File] input_gvcf_indices
+	String cohort_vcf_name
+    File scattered_calling_intervals_list
+  
+    Array[File] scattered_calling_intervals = read_lines(scattered_calling_intervals_list)
+
+    # Unzip GVCFs in parallel
+	scatter (input_gvcf in input_gvcfs) {
+
+		# Unzip GVCFs because GenotypeGVCFs is a cranky toddler
```

Comment:
I suggest stating only "Unzip GVCFs". Also, perhaps extended explanations, e.g. that GenotypeGVCFs does not handle block compressed files, should all be kept either to the task or to the workflow but not in both.